### ISSUE TITLE
Compatibility with Openfire 4.9.0

### ISSUE THREAD

**Issue by guusdk** [igniterealtime/openfire-bookmarks-plugin#20]
Version 1.1.1 (the latest release) of the Bookmarks plugin is not compatbile with the upcoming release of Openfire 4.8.0.

```
2023.09.04 18:00:52 ERROR [PluginMonitorExec-5]: org.jivesoftware.openfire.container.PluginManager - An exception occurred while loading plugin 'bookmarks':
java.lang.NoSuchMethodError: 'java.lang.String org.jivesoftware.util.JiveGlobals.getHomeDirectory()'
        at org.igniterealtime.openfire.plugin.BookmarksPlugin.getPluginJar(BookmarksPlugin.java:138) ~[bookmarks-1.1.1.jar:?]
        at org.igniterealtime.openfire.plugin.BookmarksPlugin.checkForEnterprisePlugin(BookmarksPlugin.java:82) ~[bookmarks-1.1.1.jar:?]
        at org.igniterealtime.openfire.plugin.BookmarksPlugin.initializePlugin(BookmarksPlugin.java:37) ~[bookmarks-1.1.1.jar:?]
        at org.jivesoftware.openfire.container.PluginManager.loadPlugin(PluginManager.java:637) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
        at org.jivesoftware.openfire.container.PluginMonitor$MonitorTask$4.call(PluginMonitor.java:380) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
        at org.jivesoftware.openfire.container.PluginMonitor$MonitorTask$4.call(PluginMonitor.java:368) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

**Comment by guusdk** [comment]
The Openfire 4.8.0 release candidate has been modified to avoid this issue. The `getHomeDirectory` method, based on a String argument, has been restored. This prevents the runtime issue. The restored method has been marked as being deprecated. The `Path` based API is now available under the name `getHomePath`.

This plugin should be updated to migrate to the new `Path` based API, as the deprecated API will eventually be removed.

**Comment by guusdk** [comment]
In [Openfire 4.9.0, deprecated API is being removed](https://igniterealtime.atlassian.net/browse/OF-2867) that is being used by this plugin. The deprecated API usage should be replaced, so that this plugin remains compatible with 4.9.0.

**Comment by guusdk** [comment]
Fixed by e1fa590266d416f1ecdb8ba38a6c52eb49dfcf6b
